src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py

```diff
@@ -594,15 +608,13 @@ def _predict_with_probabilities(self, input_dataset, batch_size=64,
                 all_predicted_builder.append_multiple(predicted)
                 all_probabilities_builder.append_multiple(z.tolist())
                 index += split_length 
-                if index == dataset_size - 1:
-                    done = True
 
                 cur_time = _time.time()
                 # Do not print progress if only a few samples are predicted
                 if verbose and (dataset_size >= 5
-                    and cur_time > last_time + 30 or done):
-                    print('Predicting {cur_n:{width}d}/{max_n:{width}d}'.format(
-                        cur_n = index + 1 if not done else index,
+                    and cur_time > last_time + 30 or index >= dataset_size ):
+                    print('Predicting {num_processed:{width}d}/{max_n:{width}d}'.format(
+                        num_processed = index,
```

Comment:
Has this change been testing? I'm not certain but if it's possible for this to be true `index > dataset_size - 1`, then we could be reporting that we processed more example than there are in the dataset.

Reply:
This works. 

```
model.evaluate(data)
Predicting 256/256
```

```
In [13]: model.evaluate(data)
Predicting  256/7237
Predicting 7237/7237
```